crates/apollo-compiler/src/lib.rs

```diff
@@ -1141,7 +1158,19 @@ type User
         let object_types = compiler.db.object_types();
         let object_names: Vec<_> = object_types.keys().map(|name| &**name).collect();
         assert_eq!(
-            ["Mutation", "Product", "Query", "Review", "User"],
+            [
+                "__Schema",
+                "__Type",
+                "__Field",
+                "__InputValue",
+                "__EnumValue",
+                "__Directive",
+                "Mutation",
+                "Product",
+                "Query",
+                "Review",
+                "User"
+            ],
```

Comment:
this might be good do discuss: should be explicitly exclude introspection types from `db.object_types` query? on one hand, this is useful for any sort of validation, linting, execution or future introspection work, on the other hand, this is not super useful for object type manipulation which may detract

Reply:
oh hm. my initial reaction is that it's surprising for them to be included…

I don't know if this makes sense, but my expectation would be that only explicitly requesting one of the builtins returns it:
```rust
db.object_types() // → returns Mutation, Product, Query, Review, User
db.find_object_type_by_name("__Schema") // → returns the __Schema type
```

Reply:
the implicit fields (not types) are meant to be excluded from any of the field queries, but i am wondering if having them be returned with `object_type.field("__typename")` might indeed be more useful

Reply:
all right, removed built in types from the `object_types()` query. still not 100% certain `object_type.field()` should return implicit fields 🤔 🤔 gonna think about it some more

Reply:
yea ok, let's have `object_type.field()` return implicit fields too